### PR TITLE
fix racist option names

### PR DESCRIPTION
**Pull request by jsteemann**
Docs companion PR for https://github.com/arangodb/arangodb/pull/11921

### CODE REVIEW

File: 3.8/release-notes-upgrading-changes38.md [hunk 1]
```diff
@@ -10,3 +10,17 @@ upgrading to ArangoDB 3.8, and adjust any client programs if necessary.
...
+
+The following startup options have been renamed in ArangoDB 3.8:
+
+* `--javascript.startup-options-whitelist` is now `--javascript.startup-options-allowlist`
+* `--javascript.startup-options-blacklist` is now `--javascript.startup-options-denylist`
+* `--javascript.environment-variables-whitelist` is now `--javascript.environment-variables-allowlist`
+* `--javascript.environment-variables-blacklist` is now `--javascript.environment-variables-denylist`
+* `--javascript.endpoints-whitelist` is now `--javascript.endpoints-allowlist`
+* `--javascript.endpoints-blacklist` is now `--javascript.endpoints-denylist`
+* `--javascript.files-whitelist` is now `--javascript.files-allowlist`
```
Comment (Simran-B) [inline 3.8/release-notes-upgrading-changes38.md:24]:
```suggestion
| Old name | New name |
|:---------|:---------|
| `--javascript.startup-options-whitelist` | `--javascript.startup-options-allowlist`
| `--javascript.startup-options-blacklist` | `--javascript.startup-options-denylist`
| `--javascript.environment-variables-whitelist` | `--javascript.environment-variables-allowlist`
| `--javascript.environment-variables-blacklist` | `--javascript.environment-variables-denylist`
| `--javascript.endpoints-whitelist` | `--javascript.endpoints-allowlist`
| `--javascript.endpoints-blacklist` | `--javascript.endpoints-denylist`
| `--javascript.files-whitelist` | `--javascript.files-allowlist`
```

File: 3.8/security-security-options.md [hunk 2]
```diff
@@ -91,21 +91,21 @@ in the same way as if they weren't there in first place.
...
+    --javascript.files-allowlist "^/etc/required/"
+    --javascript.files-allowlist "^/etc/mtab/"
+    --javascript.files-allowlist "^/etc/issue$"
 
 The file `/etc/issue` will be allowed to accessed and all files in the directories
 `/etc/required` and `/etc/mtab` plus their subdirectories will be accessible,
-while access to files in any other directories 
-will be disallowed from JavaScript operations, with the following exceptions:
+while access to files in any other directories will be disallowed from JavaScript 
+  operations, with the following exceptions:
```
Comment (Simran-B) [inline 3.8/security-security-options.md:108]:
```suggestion
operations, with the following exceptions:
```

**Comment by Simran-B** [comment]
Unclear if these messages should be fixed in code:
- fetching master collection dump for collection
- fetching master log from tick
- etc.

File: 3.8/architecture-deployment-modes-leader-follower-architecture.md [hunk 3]
```diff
@@ -137,7 +139,7 @@ The result might look like this:
     "ticksBehind" : 2, 
     "progress" : { 
       "time" : "2019-03-01T11:36:33Z", 
-      "message" : "fetching master log from tick 2050694546, last scanned tick 2050664547, first regular tick 2050544543, barrier: 0, open transactions: 1, chunk size 6291456", 
+      "message" : "fetching leader log from tick 2050694546, last scanned tick 2050664547, first regular tick 2050544543, barrier: 0, open transactions: 1, chunk size 6291456",
```
Comment (Simran-B) [inline 3.8/architecture-deployment-modes-leader-follower-architecture.md:142]: actually change this in code?!

File: 3.8/architecture-deployment-modes-leader-follower-architecture.md [hunk 4]
```diff
@@ -228,15 +230,15 @@ due to (repeated) connection problems:
     "lastError" : { 
       "errorNum" : 1400, 
       "time" : "2019-03-01T11:52:45Z", 
-      "errorMessage" : "could not connect to master at tcp://127.0.0.1:8529 for URL /_api/wal/tail?chunkSize=6291456&barrier=0&from=2101606369&lastScanned=2101606370&serverId=46402312160836&includeSystem=true&includeFoxxQueues=false: Could not connect to 'http+tcp://127.0.0.1:852..." 
+      "errorMessage" : "could not connect to leader at tcp://127.0.0.1:8529 for URL /_api/wal/tail?chunkSize=6291456&barrier=0&from=2101606369&lastScanned=2101606370&serverId=46402312160836&includeSystem=true&includeFoxxQueues=false: Could not connect to 'http+tcp://127.0.0.1:852..."
```
Comment (Simran-B) [inline 3.8/architecture-deployment-modes-leader-follower-architecture.md:233]: actually change this in code?!

File: 3.8/http/replications-replication-applier.md [hunk 5]
```diff
@@ -24,5 +24,5 @@ configuration of an ArangoDB database's replication applier.
 {% docublock get_api_replication_applier_state %}
 
 <!-- arangod/RestHandler/RestReplicationHandler.cpp -->
-{% docublock put_api_replication_makeSlave %}
+{% docublock put_api_replication_make_follower %}
```
Comment (Simran-B) [inline 3.8/http/replications-replication-applier.md:27]: Needs code change, requires example rebuilding

File: 3.8/monitoring-collectd-follower-status.md [hunk 6]
```diff
@@ -78,7 +79,7 @@ A running replication will return something like this:
     "safeResumeTick": "1150610894145",
     "progress": {
       "time": "2015-11-02T13:49:56Z",
-      "message": "fetching master log from tick 1150610894145",
+      "message": "fetching leader log from tick 1150610894145",
```
Comment (Simran-B) [inline 3.8/monitoring-collectd-follower-status.md:82]: actually change code?!

File: 3.8/monitoring-collectd-follower-status.md [hunk 7]
```diff
@@ -99,12 +100,12 @@ A running replication will return something like this:
 }
 ```
 
-We create a simple collectd configuration in `/etc/collectd/collectd.conf.d/slave_testDatabase.conf` that matches our API:
+We create a simple collectd configuration in `/etc/collectd/collectd.conf.d/follower_testDatabase.conf` that matches our API:
```
Comment (Simran-B) [inline 3.8/monitoring-collectd-follower-status.md:103]: can this be renamed like this?

File: 3.8/release-notes-known-issues38.md [hunk 8]
```diff
@@ -55,7 +55,7 @@ Other
...
-| **Date Added:** 2019-05-24 <br> **Component:** Web UI <br> **Deployment Mode:** Active Failover <br> **Description:** The web interface shows a wrong replication mode in the replication tab in Active Failover deployments sometimes. It may display Master/Slave mode (the default value) because of timeouts if `/_api/cluster/endpoints` is requested too frequently. <br> **Affected Versions:** 3.5.x, 3.6.x, 3.7.x <br> **Fixed in Versions:** - <br> **Reference:** N/A |
+| **Date Added:** 2019-05-24 <br> **Component:** Web UI <br> **Deployment Mode:** Active Failover <br> **Description:** The web interface shows a wrong replication mode in the replication tab in Active Failover deployments sometimes. It may display Leader/Follower mode (the default value) because of timeouts if `/_api/cluster/endpoints` is requested too frequently. <br> **Affected Versions:** 3.5.x, 3.6.x, 3.7.x <br> **Fixed in Versions:** - <br> **Reference:** N/A |
```
Comment (Simran-B) [inline 3.8/release-notes-known-issues38.md:58]: needs change in Web UI

**Comment by jsteemann** [comment]
the terms "master" and "slave" will be addressed by a seperate PR.

**Comment by Simran-B** [comment]
This PR only covers the renaming of blacklist to denylist, whitelist to allowlist and avoids the term sanity.
For additional changes, see #527

**Pull request merged**